core/include/scipp/core/element/util.h

```diff
@@ -111,4 +111,12 @@ constexpr auto fill =
     overloaded{arg_list<double, float, std::tuple<float, double>>,
                [](auto &x, const auto &value) { x = value; }};
 
+template <typename OP> struct assign_op : OP {
+  template <typename OUT, typename... IN>
+  void operator()(OUT &out, IN &&... in) {
+    out = OP::operator()(std::forward<IN>(in)...);
```

Comment:
I think this does indeed cover most cases where we have a single out argument.
should we use std::decay, to ensure OUT type is what we expect?

Reply:
Not sure what you are referring to? Aside from that, as it is `OUT` is guaranteed to be the same as `IN` (apart from `const` and `&` or `&&`), see below.